harness/tests/custom_search_mocks.py

```diff
@@ -0,0 +1,140 @@
+import logging
+
+from determined import searcher
+from determined.common.api import bindings
+from determined.experimental import client
+
+from pathlib import Path
+from typing import Any, Dict, Iterable, List, Optional, Sequence, Union
+from unittest.mock import Mock
+
+
+class SimulateMaster:
+    def __init__(self, metric: Union[float, Dict[str, Any]]) -> None:
+        self.events_queue: List[bindings.v1SearcherEvent] = []  # store event and
+        self.events_count = 0
+        self.metric = metric
+        self.overall_progress = 0.0
+
+    def handle_post_operations(
+        self, event: bindings.v1SearcherEvent, operations: List[searcher.Operation]
+    ) -> None:
+        self._remove_upto(event)
+        self._process_operations(operations)
+
+    def _remove_upto(self, event: bindings.v1SearcherEvent) -> None:
+        for i, e in enumerate(self.events_queue):
+            if e.id == event.id:
+                self.events_queue = self.events_queue[i + 1 :]
+                return
+
+        raise RuntimeError(f"event not found in events queue: {event}")
+
+    def _process_operations(self, operations: List[searcher.Operation]) -> None:
+        for op in operations:
+            self._append_events_for_op(op)  # validate_after returns two events.
+
+    def handle_get_events(self) -> Optional[Sequence[bindings.v1SearcherEvent]]:
+        return self.events_queue
+
+    def _append_events_for_op(self, op: searcher.Operation) -> None:
+        if type(op) == searcher.ValidateAfter:
```

Comment:
Nit to maybe change later: `isinstance` instead of `type(...) == `

Reply:
I did this in [6555](https://github.com/determined-ai/determined/commit/07e62b94669c111308105af383a793f253438cbd)